include/oneapi/tbb/concurrent_queue.h

```diff
@@ -215,6 +251,16 @@ class concurrent_queue {
 
     queue_allocator_type my_allocator;
     queue_representation_type* my_queue_representation;
+
+    friend bool operator==( const concurrent_queue<T, Allocator>& lhs,
+                            const concurrent_queue<T, Allocator>& rhs ){
+      return lhs.my_queue_representation == rhs.my_queue_representation;
```

Comment:
Such an implementation seems incorrect. I guess two queues are considered equal if they contains the same elements, e.g.
```
   concurrent_queue<int> q1, q2;
   q1.push(1);
   q2.push(1);
   bool expect_true = (q1 == q2);
```

`q1` and `q2` both contains only one element equal to `1`, so these queues are considered equal. But they does not point to the same representation, so your implementation of `operator==` would return `false`. I believe that we should not have the case when two queues points to the same representation - it can lean in double-free problems.